Anagram/Anagram.cs

```diff
@@ -0,0 +1,86 @@
+            this.wordList = new String[] { };
+            this.matchedWordsListFinal = new List<string>();
+
+        }
+
+        public int WordLoad()
+        {
+            this.wordList = System.IO.File.ReadAllText(@"C:\Users\Krasskoo\Desktop\WordList.txt").Split(new char[] { '.', '?', '!', ' ', ';', ':', ',', '\n' }, StringSplitOptions.RemoveEmptyEntries);
+            return this.wordList.Length;
+        }
+        public int FillMatchedWords(int totalWords)
+        {
+            String inputString = inputWordBox.Text;
+            int inputStringLength = inputString.Length;
+
+            char[] inputStringChars = inputString.ToUpper().ToCharArray();
+            Array.Sort(inputStringChars);
+
+            String sortedInputString = new string(inputStringChars);
+            for (int m = 0; m < totalWords; m++)
+            {
+                String comparedWord = this.wordList[m];
+
+                if (inputStringLength == comparedWord.Length)
+                {
+                    char[] comparedWordChars = comparedWord.ToUpper().ToCharArray();
+                    Array.Sort(comparedWordChars);
+                    String sortedComparedWord = new String(comparedWordChars);
+                    if (sortedInputString != sortedComparedWord)
+                        continue;
+                    matchedWordsListFinal.Add(comparedWord);
+
+                }
+                else
+                {
+                    for (int n = 0; n < totalWords; n++)
+                    {
+                        int nextWordOfListLength = this.wordList[n].Length;
+                        int concatenatedWordsLength = comparedWord.Length + nextWordOfListLength;
+
+                        if (nextWordOfListLength == 0 || comparedWord.Length == 0)
+                            continue;
+                        if (inputStringLength != concatenatedWordsLength)
+                            continue;
+                        char[] comparedWordChar = comparedWord.ToUpper().ToCharArray();
+                        char[] nextWordOfList = this.wordList[n].ToUpper().ToCharArray();
+                        List<char> concatenatedLengths = new List<char>(comparedWordChar.Length + nextWordOfList.Length);
+                        concatenatedLengths.AddRange(comparedWordChar);
+                        concatenatedLengths.AddRange(nextWordOfList);
+                        concatenatedLengths.ToArray();
+                        char[] concatenatedChars = concatenatedLengths.ToArray();
+                        Array.Sort(concatenatedChars);
+                        String concatenatedComparedWord = new string(concatenatedChars);
+
+                        if (sortedInputString != concatenatedComparedWord)
+                            continue;
+                        matchedWordsListFinal.Add(comparedWord + " + " + this.wordList[n]);
+                    }
+                }
+            }
+            return this.matchedWordsListFinal.Count;
```

Comment:
who cares about the count?

Reply:
None, but so I can write a test for this method as well

Reply:
this should sound wrong to you: modifying the code so that the test can be run sounds wrong. you have a class (no functional programming) so there is a better way to test the outcome. more over if you leave it like this the method becomes ambiguous: it has a side effect changing the state, the name however says nothing of what it returns, and we ACTUALLY don't care about the count. 
Nope, we don't do this:
 * the code should NEVER depend on tests! The code does not care if it's tested or not
 * the method name should always tell the user what it does and return. hidden logic is not appropriate and may lead to problems (however it's not forbidden in some specific cases)
 * keep your interfaces as simple as possible. unless developing a library do not expose more than you need.